crates/wasmtime/src/trampoline.rs

```diff
@@ -41,15 +41,18 @@ fn create_handle(
         let module = Arc::new(module);
         let runtime_info =
             &BareModuleInfo::maybe_imported_func(module, one_signature).into_traitobj();
-        let handle = OnDemandInstanceAllocator::new(config.mem_creator.clone(), 0)
-            .allocate_module(InstanceAllocationRequest {
-                imports,
-                host_state,
-                store: StorePtr::new(store.traitobj()),
-                runtime_info,
-                wmemcheck: false,
-                pkey: None,
-            })?;
+        #[cfg(feature = "async")]
+        let allocator = OnDemandInstanceAllocator::new(config.mem_creator.clone(), None, 0);
+        #[cfg(not(feature = "async"))]
+        let allocator = OnDemandInstanceAllocator::new(config.mem_creator.clone());
```

Comment:
Instead of changing the constructor of the `OnDemandInstanceAllocator` depending on whether `#[cfg(feature = "async")]`, how about we always leave out the stack creator, but then have a method for configuring it after construction:

```rust
impl OnDemandInstanceAllocator {
    #[cfg(feature = "async")]
    pub fn set_stack_creator(&mut self, stack_creator: Option<Arc<dyn StackCreator>>) {
        self.stack_creator = stack_creator;
    }
}
```

This should make construction sites a little bit simpler.